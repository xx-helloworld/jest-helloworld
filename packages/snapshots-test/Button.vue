<template>
  <button @click="handleClick">
    <span v-if="$slots.default">
      <slot></slot>
    </span>
  </button>
</template>

<script>
export default {
  name: "ElButton",
  methods: {
    handleClick(evt) {
      this.$emit("click", evt);
    }
  }
};
</script>

<style>
</style>
