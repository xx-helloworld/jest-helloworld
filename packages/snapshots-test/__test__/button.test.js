import { mount } from "@vue/test-utils";
import Button from "../Button.vue";

describe("Button", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Button, {
      slots: {
        default: "开始"
      }
    });
  });

  test("is a vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("snapshot test", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
