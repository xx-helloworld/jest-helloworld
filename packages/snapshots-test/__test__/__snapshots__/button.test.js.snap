// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button snapshot 1`] = `
<button>
  <span>
    开始
  </span>
</button>
`;

exports[`Button is a Vue instance 1`] = `
<button>
  <span>
    开始
  </span>
</button>
`;

exports[`Button snapshot test 1`] = `
<button>
  <span>
    开始
  </span>
</button>
`;
